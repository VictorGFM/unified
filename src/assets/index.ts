import AppLogo from './icons/AppLogo.svg'
import HomeIcon from './icons/HomeIcon.svg'
import GroupsIcon from './icons/GroupsIcon.svg'
import ContactsIcon from './icons/ContactsIcon.svg'
import MessagesIcon from './icons/MessagesIcon.svg'
import NotificationsIcon from './icons/NotificationsIcon.svg'
import ProfilePhotoIcon from './icons/ProfilePhotoIcon.svg'
import AppIcon from './icons/AppIcon.svg'
import InstagramIcon from './icons/InstagramIcon.svg'
import TwitterIcon from './icons/TwitterIcon.svg'
import FacebookIcon from './icons/FacebookIcon.svg'
import ToggleSocialBarIcon from './icons/ToggleSocialBarIcon.svg'

export {
  AppLogo,
  HomeIcon,
  GroupsIcon,
  ContactsIcon,
  MessagesIcon,
  NotificationsIcon,
  ProfilePhotoIcon,
  AppIcon,
  InstagramIcon,
  TwitterIcon,
  FacebookIcon,
  ToggleSocialBarIcon,
}
