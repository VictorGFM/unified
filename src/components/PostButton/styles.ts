import styled from 'styled-components'
import { COLORS } from '../../styles/global'

export const PostButtonDiv = styled.div`
  display: flex;
  align-items: center;
`

export const Button = styled.div`
  position: relative;

  width: 30px;
  height: 30px;
  margin-right: 3px;

  cursor: pointer;
  border-radius: 100%;

  &:hover {
    background-color: ${COLORS.PURPLE.TWO};
    img {
      filter: invert(44%) sepia(88%) saturate(6134%) hue-rotate(266deg) brightness(94%)
        contrast(97%);
    }
  }
`

export const ButtonImage = styled.img`
  position: absolute;
  left: 5px;
  top: 7px;

  width: 20px;
  height: 17px;
`

export const NumberInteractions = styled.span`
  font-size: 14px;
  line-height: 15px;
`
