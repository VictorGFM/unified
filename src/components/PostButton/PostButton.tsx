import { Button, ButtonImage, PostButtonDiv, NumberInteractions } from './styles'

interface PostButtonProps {
  buttonImage: string
  numberInteractions: number
}

const PostButton = ({ buttonImage, numberInteractions }: PostButtonProps) => {
  return (
    <PostButtonDiv>
      <Button>
        <ButtonImage src={buttonImage} />
      </Button>
      <NumberInteractions>{numberInteractions}</NumberInteractions>
    </PostButtonDiv>
  )
}

export default PostButton
