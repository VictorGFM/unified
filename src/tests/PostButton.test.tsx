import { describe, expect, it } from 'vitest'
import { render } from '@testing-library/react'
import PostButton from '../components/PostButton'
import { LikeIcon } from '../assets'

describe('<PostButton/>', () => {
  it('should show match snapshot', () => {
    const component = render(<PostButton buttonImage={LikeIcon} numberInteractions={5} />)

    expect(component).toMatchSnapshot()
  })
})
