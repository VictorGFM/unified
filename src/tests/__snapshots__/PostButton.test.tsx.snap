// Vitest Snapshot v1

exports[`<PostButton/> > should show match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bczRLJ kfAyjk"
      >
        <div
          class="sc-gsnTZi fHNkQO"
        >
          <img
            class="sc-dkzDqf KaWxl"
            src="/src/assets/icons/LikeIcon.svg"
          />
        </div>
        <span
          class="sc-hKMtZM esTaXM"
        >
          5
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bczRLJ kfAyjk"
    >
      <div
        class="sc-gsnTZi fHNkQO"
      >
        <img
          class="sc-dkzDqf KaWxl"
          src="/src/assets/icons/LikeIcon.svg"
        />
      </div>
      <span
        class="sc-hKMtZM esTaXM"
      >
        5
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
